### PR TITLE
Add failing test fixture for PropertyTypeDeclarationRector

### PR DESCRIPTION
**Pull request by staabm**
when a array contains a simple string, it is beeing infered to be a class-string.

for strings without a namespace the chance its really a class-name is rather low IMO.
I would expect rector to not infer a `class-string` from a pretty simple literal string like `Rocket`

# Failing Test for PropertyTypeDeclarationRector

Based on https://getrector.org/demo/1ebd59b3-d072-66a0-a8d5-ad41b9d46677

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_simple_classnames.php.inc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector\Fixture;
+
+class Rocket {}
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector\Fixture;
```
Comment (staabm) [inline rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_simple_classnames.php.inc]:
```suggestion
```

File: rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_simple_classnames.php.inc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector\Fixture;
```
[the remark this answers is not in the document]
Comment (staabm) [reply rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_simple_classnames.php.inc]: I had the impression the namespaces within the test-fixtures are required to make sure the class-name cannot collide with other tests/fixtures?

[the remark this answers is not in the document]
Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_simple_classnames.php.inc]: It did make collision, but as far as I know, it now no longer make collision.

It is still recommended to has namespace for fixture, but for specific purpose like this PR, it is now safe to not has a namespace.

**Comment by TomasVotruba** [comment]
I'm looking at test now and it seems processed correctly:

image

**Comment by staabm** [comment]
just rebased the PR. it seems still having the initial reported problem:

grafik

**Comment by TomasVotruba** [comment]
I see. We depend on PHPStan here, so we have to override the behavior.

You can re-use simple behavior to remove `class-string` to `string`:

* `Rector\TypeDeclaration\TypeAnalyzer\GenericClassStringTypeNormalizer::normalize()`

**Comment by samsonasik** [comment]
@staabm I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/501

**Pull request closed without merging**